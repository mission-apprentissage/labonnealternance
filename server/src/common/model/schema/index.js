import { apiCallSchema } from "./ApiCall.js"
import { applicationSchema } from "./applications.js"
import { appointmentSchema } from "./appointment.js"
import { credentialSchema } from "./credentials.js"
import { diplomesMetiersSchema } from "./diplomesmetiers.js"
import { domainesMetiersSchema } from "./domainesmetiers.js"
import { etablissementSchema } from "./etablissement.js"
import { formulaireSchema } from "./formulaire.js"
import { bonneBoiteSchema } from "./lbb/bonneBoite.js"
import { emailBlacklist } from "./lbb/emailBlacklist.js"
import { geoLocationSchema } from "./lbb/geoLocations.js"
import { opco } from "./lbb/opco.js"
import { mnaFormationSchema } from "./mnaFormation/mnaFormation.js"
import { offreSchema } from "./offre.js"
import { optoutSchema } from "./optout.js"
import { parcoursupEtablissementStatSchema } from "./parcoursupEtablissementStat.js"
import { userRecruteurSchema } from "./recruteur/users.js"
import { referentielOpcoSchema } from "./referentielOpco.js"
import { userSchema } from "./user.js"
import { widgetParameterSchema } from "./widgetParameter.js"

export default {
  domainesMetiersSchema,
  diplomesMetiersSchema,
  applicationSchema,
  apiCallSchema,
  mnaFormationSchema,
  bonneBoiteSchema,
  geoLocationSchema,
  emailBlacklist,
  opco,
  userSchema,
  appointmentSchema,
  widgetParameterSchema,
  etablissementSchema,
  parcoursupEtablissementStatSchema,
  formulaireSchema,
  offreSchema,
  referentielOpcoSchema,
  optoutSchema,
  userRecruteurSchema,
  credentialSchema,
}
