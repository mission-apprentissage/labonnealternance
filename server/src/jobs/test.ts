import { ObjectId } from "bson"
import { IApplicant, IApplication, ZApplicant } from "shared"

import { getDbCollection } from "@/common/utils/mongodbUtils"

import { asyncForEach } from "../common/utils/asyncUtils"

type ApplicationAggregate = Pick<IApplication, "applicant_email" | "applicant_first_name" | "applicant_last_name" | "applicant_phone" | "_id">

export const migrationTest = async () => {
  // console.log(extensions.telephone.safeParse("0690760552"))
  // return
  const applications = (await getDbCollection("applications")
    .find({}, { projection: { applicant_first_name: 1, applicant_last_name: 1, applicant_email: 1, applicant_phone: 1, _id: 1 } })
    .toArray()) as ApplicationAggregate[]

  const stat = { error: 0, success: 0, total: applications.length }

  await asyncForEach(applications, async (application) => {
    const { applicant_first_name, applicant_email, applicant_last_name, applicant_phone, _id } = application
    const now = new Date()
    const applicant: IApplicant = {
      _id: new ObjectId(),
      firstname: applicant_first_name,
      lastname: applicant_last_name,
      phone: applicant_phone,
      email: applicant_email,
      last_connection: null,
      createdAt: now,
      updatedAt: now,
    }
    const validation = ZApplicant.safeParse(applicant)
    if (!validation.success) {
      stat.error++
      console.log({ applicant, err: validation.error.errors })
    } else {
      stat.success++
    }
  })
  console.log(stat)
}
