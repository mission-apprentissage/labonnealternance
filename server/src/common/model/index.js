import mongoosePaginate from "mongoose-paginate"
import { getElasticInstance, mongoosastic } from "../esClient/index.js"
import schema from "../model/schema/index.js"
import { mongooseInstance } from "../mongodb.js"

const createModel = (modelName, descriptor, options = {}) => {
  const schema = new mongooseInstance.Schema(descriptor)

  if (options.esIndexName) {
    schema.plugin(mongoosastic, { esClient: getElasticInstance(), index: options.esIndexName })
  }

  if (options.esIndexName || options.paginate) {
    schema.plugin(mongoosePaginate)
  }

  if (options.createMongoDBIndexes) {
    options.createMongoDBIndexes(schema)
  }

  return mongooseInstance.model(modelName, schema)
}

const DomainesMetiers = createModel("domainesmetiers", schema.domainesMetiersSchema, {
  esIndexName: "domainesmetiers",
})
const FormationCatalogue = createModel("formationcatalogues", schema.mnaFormationSchema, {
  esIndexName: "formationcatalogues",
})
const DiplomesMetiers = createModel("diplomesmetiers", schema.diplomesMetiersSchema, {
  esIndexName: "diplomesmetiers",
})
const ApiCalls = createModel("apicalls", schema.apiCallSchema)
const Application = createModel("applications", schema.applicationSchema, {
  paginate: true,
})
const SourceFormations = createModel("sourceformations", schema.sourceFormationsSchema)
const GeoLocation = createModel("geolocation", schema.geoLocationSchema)
const EmailBlacklist = createModel("emailblacklist", schema.emailBlacklist)
const Opco = createModel("opco", schema.opco)
const BonnesBoites = createModel("bonnesboites", schema.bonneBoiteSchema, {
  esIndexName: "bonnesboites",
})
const User = createModel("user", schema.userSchema, { paginate: true })
const Appointment = createModel("appointment", schema.appointmentSchema, { paginate: true })
const WidgetParameter = createModel("widgetParameter", schema.widgetParameterSchema, { paginate: true })
const Etablissement = createModel("etablissement", schema.etablissementSchema, { paginate: true })
const ParcoursupEtablissementStat = createModel("parcoursupEtablissementStat", schema.parcoursupEtablissementStatSchema, { paginate: true })
const Formulaire = createModel("formulaire", schema.formulaireSchema, { esIndexName: "formulaires", paginate: true })
const Credential = createModel("credentials", schema.credentialSchema)
const Offre = createModel("offre", schema.offresSchema)
const Optout = createModel("optout", schema.optoutSchema, { paginate: true })
const ReferentielOpco = createModel("referentielOpco", schema.referentielOpcoSchema)
const UserRecruteur = createModel("userRecruteur", schema.userRecruteurSchema, { paginate: true })
const AppointmentDetailed = createModel("appointmentDetailed", schema.appointmentSchema)

export {
  DomainesMetiers,
  FormationCatalogue,
  DiplomesMetiers,
  ApiCalls,
  Application,
  SourceFormations,
  GeoLocation,
  EmailBlacklist,
  Opco,
  BonnesBoites,
  User,
  Appointment,
  WidgetParameter,
  Etablissement,
  ParcoursupEtablissementStat,
  Formulaire,
  Credential,
  Offre,
  Optout,
  ReferentielOpco,
  UserRecruteur,
  AppointmentDetailed,
}
