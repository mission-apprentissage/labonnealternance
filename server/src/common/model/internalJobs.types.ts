import { ObjectId } from "mongodb"

import { CronName } from "@/jobs/jobs"

/*
  name: Le nom de la tâche
  type: Type du job simple ou cron
  status: Statut courant du job
  sync: Si le job est synchrone
  payload: La donnée liéé à la tâche
  output: Les valeurs de retours du job
  cron_string: standard cron string exemple: '*\/2 * * * *'
  scheduled_for: Date de lancement programmée
  started_at: Date de lancement
  ended_at: Date de fin d'execution
  updated_at: Date de mise à jour en base de données
  created_at: Date d'ajout en base de données
*/

interface IInternalJobsSimple {
  _id: ObjectId
  name: string
  type: "simple"
  status: "pending" | "will_start" | "running" | "finished" | "blocked" | "errored"
  sync: boolean
  payload: Record<string, any>
  output?: Record<string, unknown> | undefined
  scheduled_for: Date
  started_at?: Date
  ended_at?: Date
  updated_at: Date
  created_at: Date
}

interface IInternalJobsCron {
  _id: ObjectId
  name: CronName
  type: "cron"
  status: "pending" | "will_start" | "running" | "finished" | "blocked" | "errored"
  sync: boolean
  cron_string: string
  scheduled_for: Date
  updated_at: Date
  created_at: Date
}

interface IInternalJobsCronTask {
  _id: ObjectId
  name: CronName
  type: "cron_task"
  status: "pending" | "will_start" | "running" | "finished" | "blocked" | "errored"
  sync: boolean
  scheduled_for: Date
  started_at?: Date
  ended_at?: Date
  updated_at: Date
  created_at: Date
}

type IInternalJobs = IInternalJobsSimple | IInternalJobsCron | IInternalJobsCronTask

export type { CronName, IInternalJobs, IInternalJobsCron, IInternalJobsCronTask, IInternalJobsSimple }
