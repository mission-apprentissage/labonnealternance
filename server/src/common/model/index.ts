import { captureException } from "@sentry/node"

import { db, mongooseInstance } from "@/common/mongodb"

import { logger } from "../logger"

import AnonymizedUser from "./schema/anonymizedUsers/anonymizedUsers.schema"
import ApiCalls from "./schema/apiCall/apiCall.schema"
import AnonymizedApplication from "./schema/application/anonymizedApplications.schema"
import Application from "./schema/application/applications.schema"
import AppointmentDetailed from "./schema/appointmentDetailed/appointmentDetailed.schema"
import Appointment from "./schema/appointments/appointment.schema"
import Credential from "./schema/credentials/credential.schema"
import CustomEmailETFA from "./schema/customEmailETFA/customEmailETFA.schema"
import DiplomesMetiers from "./schema/diplomesmetiers/diplomesmetiers.schema"
import DomainesMetiers from "./schema/domainesmetiers/domainesmetiers.schema"
import EligibleTrainingsForAppointment from "./schema/eligibleTrainingsForAppointment/eligibleTrainingsForAppointment.schema"
import eligibleTrainingsForAppointmentHistory from "./schema/eligibleTrainingsForAppointmentsHistory/eligibleTrainingsForAppointmentHistory.schema"
import EmailBlacklist from "./schema/emailBlacklist/emailBlacklist.schema"
import Etablissement from "./schema/etablissements/etablissement.schema"
import FicheMetierRomeV4 from "./schema/ficheRomeV4/ficheRomeV4"
import FormationCatalogue from "./schema/formationCatalogue/formationCatalogue.schema"
import GeoLocation from "./schema/geolocation/geolocation.schema"
import InternalJobs from "./schema/internalJobs/internalJobs.schema"
import Job from "./schema/jobs/jobs.schema"
import LbaCompany from "./schema/lbaCompany/lbaCompany.schema"
import LbaCompanyLegacy from "./schema/lbaCompanylegacy/lbaCompanyLegacy.schema"
import { Cfa } from "./schema/multiCompte/cfa.schema"
import { Entreprise } from "./schema/multiCompte/entreprise.schema"
import { RoleManagement } from "./schema/multiCompte/roleManagement.schema"
import { UserWithAccount } from "./schema/multiCompte/userWithAccount.schema"
import Opco from "./schema/opco/opco.schema"
import Optout from "./schema/optout/optout.schema"
import Recruiter from "./schema/recruiter/recruiter.schema"
import RecruteurLbaUpdateEvent from "./schema/recruteurLbaUpdateEvent/recruteurLbaUpdateEvent.schema"
import ReferentielOnisep from "./schema/referentielOnisep/referentielOnisep.schema"
import ReferentielOpco from "./schema/referentielOpco/referentielOpco.schema"
import ReferentielRome from "./schema/referentielRome/referentielRome"
import Session from "./schema/session/session.schema"
import SiretDiffusibleStatus from "./schema/siretDiffusibleStatusSchema/siretDiffusibleStatusSchema.schema"
import UnsubscribedLbaCompany from "./schema/unsubscribedLbaCompany/unsubscribedLbaCompany.schema"
import UnsubscribeOF from "./schema/unsubscribedOF/unsubscribeOF.schema"
import User from "./schema/user/user.schema"

const createSpecialIndexes = async () => {
  await db.collection("bonnesboites").createIndex({ geopoint: "2dsphere" })
  await db.collection("formationcatalogues").createIndex({ lieu_formation_geopoint: "2dsphere" })
  await db.collection("recruiters").createIndex({ geopoint: "2dsphere" })
}

export async function createMongoDBIndexes() {
  const results = await Promise.allSettled(
    mongooseInstance.modelNames().map(async (name) => {
      const model = mongooseInstance.model(name)
      return model.createIndexes({ background: true }).catch(async (e) => {
        if (e.codeName === "IndexOptionsConflict" || e.codeName === "IndexKeySpecsConflict") {
          const err = new Error(`Conflict in indexes for ${name}`, { cause: e })
          logger.error(err)
          captureException(err)
          await mongooseInstance.connection.collection(model.collection.name).dropIndexes()
          await model.createIndexes({ background: true })
        } else {
          throw e
        }
      })
    })
  )

  await createSpecialIndexes()

  const errors = results.reduce((acc, r) => {
    if (r.status === "rejected") {
      acc.push(r.reason)

      logger.error(r.reason)
      captureException(r.reason)
    }

    return acc
  }, [] as Error[])

  if (errors.length > 0) {
    throw new AggregateError(errors, `createMongoDBIndexes failed with ${errors.length} errors`)
  }
}

export {
  AnonymizedApplication,
  AnonymizedUser,
  ApiCalls,
  Application,
  Appointment,
  AppointmentDetailed,
  Cfa,
  Credential,
  CustomEmailETFA,
  DiplomesMetiers,
  DomainesMetiers,
  EligibleTrainingsForAppointment,
  EmailBlacklist,
  Entreprise,
  Etablissement,
  FicheMetierRomeV4,
  FormationCatalogue,
  GeoLocation,
  InternalJobs,
  Job,
  LbaCompany,
  LbaCompanyLegacy,
  RecruteurLbaUpdateEvent,
  Opco,
  Optout,
  Recruiter,
  ReferentielOnisep,
  ReferentielOpco,
  ReferentielRome,
  RoleManagement,
  Session,
  SiretDiffusibleStatus,
  UnsubscribeOF,
  UnsubscribedLbaCompany,
  User,
  UserWithAccount,
  eligibleTrainingsForAppointmentHistory,
}
