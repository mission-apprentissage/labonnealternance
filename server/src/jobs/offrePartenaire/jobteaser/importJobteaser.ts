import { Readable } from "stream"
import { createGunzip } from "zlib"

import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import rawJobteaserModel from "shared/models/rawJobteaser.model"

import { importFromStreamInXml } from "@/jobs/offrePartenaire/importFromStreamInXml"
import { jobteaserJobToJobsPartners, ZJobteaserJob } from "@/jobs/offrePartenaire/jobteaser/jobteaserMapper"

import config from "../../../config"
import { rawToComputedJobsPartners } from "../rawToComputedJobsPartners"

const rawCollectionName = rawJobteaserModel.collectionName
const offerXmlTag = "job"

type ListType = { last_modified: string; mode: number; name: string; size: number }
export const importJobteaserRaw = async (sourceStream?: NodeJS.ReadableStream) => {
  if (sourceStream) {
    await importFromStreamInXml({
      destinationCollection: rawCollectionName,
      offerXmlTag,
      stream: sourceStream,
      partnerLabel: JOBPARTNERS_LABEL.JOBTEASER,
      conflictingOpeningTagWithoutAttributes: true,
    })
  } else {
    // fetch token
    const tokenResponse = await fetch("https://partneraccess.jobteaser.com/api/v2/user/token", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Basic ${btoa(`${config.jobteaserUserName}:${config.jobteaserPassword}`)}`,
      },
    })
    const token = ((await tokenResponse.json()) as { access_token: string; expires_at: string }).access_token

    // fetch list of files
    const listResponse = await fetch("https://partneraccess.jobteaser.com/api/v2/user/dirs", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
    })
    const list: ListType[] = (await listResponse.json()) as ListType[]
    const latestFile = list.at(-1)?.name

    // fetch latest file
    const fileResponse = await fetch(`https://partneraccess.jobteaser.com/api/v2/user/files?path=${latestFile}`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${token}`,
      },
    })

    const nodeStream = Readable.fromWeb(fileResponse.body!)
    const gunzip = createGunzip()
    const fileSourceStream = nodeStream.pipe(gunzip)
    await importFromStreamInXml({
      destinationCollection: rawCollectionName,
      offerXmlTag,
      stream: fileSourceStream,
      partnerLabel: JOBPARTNERS_LABEL.JOBTEASER,
      conflictingOpeningTagWithoutAttributes: true,
    })
  }
}

export const importJobteaserToComputed = async () => {
  await rawToComputedJobsPartners({
    collectionSource: rawCollectionName,
    partnerLabel: JOBPARTNERS_LABEL.JOBTEASER,
    zodInput: ZJobteaserJob,
    mapper: jobteaserJobToJobsPartners,
    documentJobRoot: offerXmlTag,
  })
}
