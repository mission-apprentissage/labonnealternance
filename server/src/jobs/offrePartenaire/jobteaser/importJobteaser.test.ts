import fs from "node:fs"

import { omit } from "lodash-es"
import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import { beforeEach, describe, expect, it, vi } from "vitest"

import { getDbCollection } from "@/common/utils/mongodbUtils"
import { useMongo } from "@tests/utils/mongo.test.utils"

import { importJobteaserRaw, importJobteaserToComputed } from "./importJobteaser"

const now = new Date("2024-07-21T04:49:06.000+02:00")

describe("importJobteaser", () => {
  useMongo()

  beforeEach(() => {
    vi.useFakeTimers()
    vi.setSystemTime(now)

    return async () => {
      vi.useRealTimers()
      await getDbCollection("computed_jobs_partners").deleteMany({})
      await getDbCollection("raw_jobteaser").deleteMany({})
    }
  })

  it("should test the import of Jobteaser data into computed_job_partners", async () => {
    const fileStream = fs.createReadStream("server/src/jobs/offrePartenaire/jobteaser/importJobteaser.test.input.xml")
    await importJobteaserRaw(fileStream)
    expect.soft(await getDbCollection("raw_jobteaser").countDocuments({})).toBe(2)

    await importJobteaserToComputed()
    const jobs = (
      await getDbCollection("computed_jobs_partners")
        .find({ partner_label: JOBPARTNERS_LABEL.JOBTEASER }, { projection: { _id: 0, created_at: 0 } })
        .toArray()
    ).sort((a, b) => ((a.partner_job_id ?? "") < (b.partner_job_id ?? "") ? -1 : 1))
    const filtered = jobs.map((x) => omit(x, ["offer_creation", "offer_expiration"]))
    expect.soft(jobs.length).toBe(2)
    expect.soft(filtered).toMatchSnapshot()
  })
})
