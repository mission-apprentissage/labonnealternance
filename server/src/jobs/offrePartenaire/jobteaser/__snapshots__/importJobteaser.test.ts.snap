// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`importJobteaser > should test the import of Jobteaser data into computed_job_partners 1`] = `
[
  {
    "apply_phone": null,
    "apply_url": "https://www.jobteaser.com/fr/job-offers/ab3c7310-98eb-4541-9a91-438dafddfae3-schneider-electric-alternance-webmarketing-specialist-bac-4-5-38-grenoble-h-f?utm_campaign=partner_access&utm_source=la_bonne_alternance",
    "business_error": null,
    "contract_duration": null,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": null,
    "contract_type": [
      "Apprentissage",
      "Professionnalisation",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-09-20T09:42:48.000Z,
    "offer_description": "<strong>Missions : </strong> <br><br>Au sein du service communication du pays France de Schneider Electric, dans le pôle campagnes digitales, l'alternant aura pour missions d'assister les chefs de projet web dans l'intégration des assets digitaux dans les différents outils. <br><br><ul><li> Réceptionner les briefs de la part des program managers / chef de projet web : respect des délais, vérification des livrables attendus, mises en place d'actions correctives si nécessaire. </li><li> Intégrer les emailings et assets de marketing automation </li><li> Intégrer / modifier les pages web </li><li> Préparer les audiences clients dans la customer data plateforme </li><li> Mesurer et analyser la performance des outils au travers des différents outils : Google Analytics... </li><li> Participer de façon active aux différents points de suivi avec les programs managers et experts digitaux : suivi des projets, challenger les parcours clients, apporter de nouvelles idées... </li></ul> <br> <strong>Profil recherché : </strong> <br><ul><li> BA ou Master ou ESC spécialité Digitale / Webmarketing / Systèmes d'informations </li><li> Première expérience significative en gestion de projet ou marketing ou marketing digital </li><li> Intérêt marqué pour le digital, impérativement orienté client </li><li> Bonne capacité d'analyse et de synthèse </li><li> Excellent relationnel et goøt du travail en équipe </li><li> Autonomie et débrouillardise, capacité d'organisation </li><li> Créativité </li><li> Expérience en agence de communication digitale, ou dans un service de marketing/communication serait un plus </li></ul><br><br><strong>Compétences : </strong> <br><ul><li> Langues : français courant (bon rédactionnel), maîtrise impérative de l'anglais parlé et écrit (exposition internationale) </li></ul><ul><li> Logiciels : maîtrise impérative de la suite Office (principalement Excel et PowerPoint) </li><li> Connaissance des outils digitaux (emailing, SEO, SEM, animation de communautés, etc) </li><li> Connaissances des technologies web est un plus (SQL/ html/css, gestion de contenu avec solution CMS) </li></ul><br><br><strong>Informations complémentaires : </strong> <br><br><ul><li> <strong>Durée de contrat : </strong> Alternance pendant 2 ans (préféré) mais possibilité d'1 an aussi.</li><li><strong>Localisation : </strong>Grenoble</li><li><strong>Démarrage</strong> : Septembre 2025 </li></ul><br><br><strong>Laissez-nous vous connaître ! Postulez dès aujourd'hui.</strong> <br><br> <strong>Vous devez soumettre une candidature en ligne pour être pris en considération pour un poste chez nous. Ce poste sera affiché jusqu'à ce qu'il soit pourvu.</strong> <br><br><strong> En intégrant Schneider Electric en tant qu'Alternant, en plus d'une expérience professionnelle et personnelle enrichissante, vous bénéficierez </strong><strong> : </strong><br><ul><li><strong> D'un package salarial attractif et compétitif </strong><strong> : </strong> salaire supérieur au minimum légal, intéressement et participation, prime de réussite à l'examen de 1000 </li><li><strong> D'une participation aux frais de transport en commun </strong> à hauteur de 75% </li><li><strong> D'une aide à la mobilité </strong><strong> : </strong> en fonction de votre situation d'éloignement, nous vous proposons une <strong>participation aux frais de transport</strong> (train/avion/voiture/bus) <strong>ou frais kilométriques</strong> <strong>ou une aide " double résidence "</strong> jusqu'à 550 par mois </li><li><strong> D'autres avantages </strong><strong> : </strong></li><ul><li> un accès au restaurant d'entreprise ou ticket restaurant pour les sites ne disposant pas de restaurant d'entreprise, </li><li> un accompagnement à la mobilité internationale si obligatoire dans le cursus, </li><li> 5 jours de congés " préparation aux examens " rémunérés pour tous, </li><li> des activités culturelles et sportives, </li><li> des dispositifs d'aides sociales si besoin pendant votre contrat. </li></ul></ul><br> et pour en savoir plus sur notre politique Diversité et Inclusion France : https://www.se.com/fr/fr/about-us/diversity-and-inclusion/ <br><br><strong> Vous cherchez à créer de l'IMPACT dans votre carrière ? </strong><br> Lorsque vous envisagez de rejoindre une nouvelle équipe, la culture d'entreprise est essentielle. Chez Schneider Electric, nos valeurs et nos comportements sont le fondement de la création d'une culture d'entreprise qui contribue à la réussite de l'entreprise. Nous pensons que nos valeurs IMPACT - Inclusion, Maîtrise, Raison d'Etre, Action, Curiosité, Travail d'équipe - commencent avec nous. <br> IMPACT est également une invitation à rejoindre une entreprise où vous pourrez contribuer à transformer l'ambition du développement durable en action, quelle que soit votre fonction dans l'entreprise. C'est un appel à associer votre carrière à l'ambition d'un monde plus résilient, plus efficace et plus durable. <br> Nous recherchons des IMPACT Makers, des personnes exceptionnelles qui transforment les ambitions de développement durable en actions à l'intersection de l'automatisation, de l'électrification et de la digitalisation. Nous célébrons les IMPACT Makers et pensons que chacun a le potentiel d'en être un. <br> Devenez un IMPACT Maker chez Schneider Electric - postulez dès aujourd'hui ! <br><br> 36 milliards d'euros de chiffre d'affaires global<br>+13% de croissance organique<br>150 000+ employés dans plus de 100 pays<br>#N&deg; 1 du classement Global 100 des entreprises les plus durables au monde <br><br>Vous devez soumettre une demande en ligne pour que votre profil soit pris en considération pour un poste chez nous. Ce poste sera visible jusqu'à ce qu'il soit pourvu. <br><br>Schneider Electric aspire à être l'entreprise la plus inclusive et la plus solidaire au monde, en offrant des opportunités équitables à chacun, partout, et en veillant à ce que tous les employés se sentent valorisés et en sécurité pour donner le meilleur d'eux-mêmes. Nous pensons que nos différences nous rendent plus forts en tant qu'entreprise et en tant qu'individus. Nous nous engageons à promouvoir l'inclusion dans tout ce que nous faisons. <br><br>Chez Schneider Electric, nous respectons les normes les plus strictes en matière d'éthique et de conformité, et nous pensons que la confiance est une valeur fondamentale. Notre Charte de Confiance est notre Code de Conduite. Elle démontre notre engagement en matière d'éthique, de sécurité, de développement durable, de qualité et de cybersécurité. Elle sous-tend chaque aspect de notre activité ainsi que notre volonté de nous comporter de manière respectueuse et de bonne foi face à toutes nos parties prenantes. Découvrez notre charte de confiance ici . <br><br> Schneider Electric est un employeur qui souscrit au principe de l'égalité des chances. Notre politique est de fournir des opportunités d'emploi et d'avancement égales dans les domaines du recrutement, de l'embauche, de la formation, de la mutation et de la promotion de toutes les personnes qualifiées, indépendamment de l'origine, de la religion, de la couleur, du sexe, du handicap, de l'ascendance, de l'âge, du statut militaire, de l'orientation sexuelle, de la situation matrimoniale ou de toute autre caractéristique ou comportement protégé par la loi.",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-20T09:42:48.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": null,
    "offer_title": "Alternance - webmarketing specialist - Bac+4/5 - (38 - Grenoble) - H/F",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "ab3c7310-98eb-4541-9a91-438dafddfae3",
    "partner_label": "Jobteaser",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": "Grenoble",
    "workplace_address_label": "Grenoble",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "Schneider Electric",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_phone": null,
    "apply_url": "https://www.jobteaser.com/fr/job-offers/d495551f-394a-462a-94e2-01ec47abfd37-valeo-alternance-ingenieur-r-d-commercial-vehicles-2025-2028?utm_campaign=partner_access&utm_source=la_bonne_alternance",
    "business_error": null,
    "contract_duration": null,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": null,
    "contract_type": [
      "Apprentissage",
      "Professionnalisation",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-09-19T23:27:00.000Z,
    "offer_description": "Valeo est une entreprise mondiale de haute technologie qui conçoit des solutions révolutionnaires pour réinventer la mobilité. Nous sommes un équipementier partenaire des constructeurs automobiles et des acteurs de la nouvelle mobilité dans le monde entier. Notre vision ? Inventer une mobilité plus verte et plus sûre, grâce à des solutions axées sur la conduite intuitive et la réduction des émissions de CO2. Nous sommes leaders dans nos domaines d'activité et reconnus comme l'une des plus grandes entreprises innovantes au monde.<br><br><b>Valeo est une entreprise mondiale de haute technologie qui conçoit des solutions révolutionnaires pour réinventer la mobilité. Nous sommes un équipementier partenaire des constructeurs automobiles et des acteurs de la nouvelle mobilité dans le monde entier. Notre vision ? Inventer une mobilité plus verte et plus sûre, grâce à des solutions axées sur la conduite intuitive et la réduction des émissions de CO2. Nous sommes leaders dans nos domaines d'activité et reconnus comme l'une des plus grandes entreprises innovantes au monde.</b><br><ul><li><b>Au sein du Groupe de Produits Systèmes de Transmission, le site de Valeo Amiens est composé d'un centre de recherche et d'un site industriel. Le centre de recherche travaille entre autres au développement de systèmes de transmission pour l'électrification des véhicules automobiles et poids lourds.</b></li><li><b>Vous êtes intégré(e) au sein du service R&D de la Plate-Forme DTTS - Commercial Vehicles</b></li></ul><br><b><b>Vos challenges ? </b></b><br><br><ul><li><b><b>Vous supportez les activités de conception des nouveaux produits de la plateforme : nouvel amortisseur 430 TH Gen2 et nouveau kit Fort Couple 430 TH Gen1 et 430 DBE</b></b></li><li><b><b>Vous participez à la réalisation et à la modification de plans CAO des composants et des produits</b></b></li><li><b><b>Vous réalisez des études avec réalisation de plans pour les services supports (prototypes et essais)</b></b></li><li><b><b>Vous réalisez la création et la mise à jour des Checks Lists Conception</b></b></li><li><b><b>Vous tenez à jour et améliorez le catalogue produits WorldWide</b></b></li><li><b><b>Vous supportez l'activité analyse de la concurrence pour l'aspect Design</b></b></li></ul><br><b><b>Parlons technique... </b></b><br><ul><li><b><b>Vous êtes inscrit(e) en Cursus d'Ingénieur Généraliste ou à dominante mécanique</b></b></li><li><b><b>Vous maîtrisez l'anglais.</b></b></li><li><b><b>Vous maîtrisez les outils informatiques pour l'aspect bureautique</b></b></li><li><b><b> Vous maîtrisez aussi la CAO avec le logiciel CATIA V5.</b></b></li><li><b><b>Vous avez un bon relationnel</b></b></li></ul><br><b><b>En plus... </b></b><br><ul><li><b><b>intérêt pour l'automobile</b></b></li><li><b><b>Rigoureux, concret, dynamique</b></b></li><li><b><b>Travail en équipe</b></b></li></ul><br><b><b>Pourquoi rejoindre notre équipe ?</b></b><br><br><b>Si vous êtes motivé(e) par cette alternance et souhaitez rejoindre une équipe dynamique dans un environnement stimulant, envoyez nous votre CV ainsi qu'une lettre de motivation via notre plateforme</b><br><br><b>Grâce à sa stratégie fondée sur l'innovation, Valeo est au cœur des trois révolutions qui bouleversent l'automobile: l'électrification du véhicule, l'avènement du véhicule autonome et connecté, et le développement de la mobilité digitale. </b><br><br><b>Des révolutions que Valeo a su anticiper et qui sont aujourd'hui synonymes de formidables opportunités de carrière, y compris à l'international ! Chez Valeo, l'innovation est portée par la diversité, l'authenticité et l'énergie de ses talents. Vous souhaitez vivre de nouvelles aventures technologiques et humaines ? Rejoignez Valeo et ses 114 700 collaborateurs à travers 33 pays !</b><br><br><b>Plus d'informations sur valeo : https://www.valeo.com</b><br><br><b>Position:</b><br>R&D Trainee/Apprentice/VIE<br><b>Département:</b><br>Mechanical Development - J<br><b>Schedule:</b><br>Temps plein<br><b>Employee Status:</b><br>Alternant (durée déterminée)<br><b>Type de contrat:</b><br>Contrat d'alternance CDD<br><b>Job Posting Date:</b><br>2025-09-19<br>Join Us !<br>Being part of our team, you will join:<br>- one of the largest global innovative companies, with more than 20,000 engineers working in Research & Development<br>- a multi-cultural environment that values diversity and international collaboration<br>- more than 100,000 colleagues in 31 countries... which make a lot of opportunity for career growth<br>- a business highly committed to limiting the environmental impact if its activities and ranked by Corporate Knights as the number one company in the automotive sector in terms of sustainable development<br><br>More information on Valeo: https://www.valeo.com",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-19T23:27:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": null,
    "offer_title": "Alternance - Ingénieur R&D Commercial Véhicles 2025-2028",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "d495551f-394a-462a-94e2-01ec47abfd37",
    "partner_label": "Jobteaser",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": "Amiens",
    "workplace_address_label": "Amiens",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "Valeo",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
]
`;
